lib/begin.go

```diff
@@ -257,3 +240,130 @@ func (a *ACBuild) beginFromRemoteImage(start string, insecure bool) error {
 
 	return util.ExtractImage(path.Join(tmpDepStoreTarPath, files[0].Name()), a.CurrentACIPath, nil)
 }
+
+func (a *ACBuild) BeginFromLocalDockerImage(start string, dockerImageName string) (err error) {
+	err = a.initDirectory()
+	if err != nil {
+		return nil
+	}
+
+	if err = a.lock(); err != nil {
+		return err
+	}
+	defer func() {
+		if err != nil {
+			os.RemoveAll(a.ContextPath)
+		}
+		if err1 := a.unlock(); err == nil {
+			err = err1
+		}
+	}()
+
+	renderedACIs, err := docker2aci.ConvertFile(dockerImageName, start, true, a.ContextPath, "")
```

Comment:
IMO de-dupe all the boilerplate around this line by having:
`type renderFunc func() (string, error)`
`func (a *ACBuild) beginFromDocker(rf renderFunc) (err error) { ... }`
```
func (a *ACBuild) BeginFromLocalDockerImage(start string, dockerImageName string) (err error) { 
    return a.beginFromDocker(func() (string, error) { 
        return docker2aci.ConvertFile(dockerImageName, start, true, a.ContextPath, "") 
    }) 
}
```
etc